OpenJ9 core dump gathered with exclusive access shows J9DDRCorruptData inspecting class fields

Java -version output
--------------------

Not sure how to get the equivalent of `java -version` from `jdmpview` but the `context` reports:

```
*0 : PID: 21516 : JRE 1.8.0 Windows 10 amd64-64 (build 1.8.0_292-b10)
```

Summary of problem
------------------

Inspecting some class fields shows `J9DDRCorruptData`:

```
> x/j 0x6845409f8
         heap #1 - name: Generational@1b11705d8a0

          java/lang/Class @ 0x6845409f8
           declared fields:
            private transient long vmRef = 3498764144 (0xd08ae770)
            private transient ClassLoader classLoader = null
            private transient java.security.ProtectionDomain protectionDomain = Problem running command:
com.ibm.j9ddr.view.dtfj.image.J9DDRCorruptData incompatible with com.ibm.dtfj.java.JavaField
```

The dump is supposed to have been taken with `exclusive+prepwalk` although my technique of verifying that using `!findcallsite` on `dmpagent.c` allocations and casting the last one to `J9RASdumpAgent` no longer seems to work as that's pointing to a Snap agent which is wrong, so I'm not sure how to verify that.

The dump is in an exclusive access state:

```
> !j9javavm 0x1b11707d010 | grep exclusiveAccessState
        0x1ae0: U64 exclusiveAccessState = 0x0000000000000002 (2)
```

Diagnostic files
----------------

`S:\sf\TS006\232\TS006232802\2021-09-03\ojdk8_defaultServer.zip_unpack\defaultServer\core.20210903.095600.21516.0001.dmp`

Here's another issue running with the latest OpenJ9 `jdmpview` (also seen in MAT):

```
> x/j 0x684540db0
         heap #1 - name: Generational@1b11705d8a0

          [C @ 0x684540db0
           <cannot determine if above object is array (<corrupt data encountered>); we will assume it is not an array>

            references: <none>
```

Full details of the corrupt data exception from MAT:

```
com.ibm.j9ddr.view.dtfj.DTFJCorruptDataException: J9DDRCorruptData [as=minidump : 0 Message: Internal runtime exception was encountered]
        at com.ibm.j9ddr.view.dtfj.J9DDRDTFJUtils.handleAsCorruptDataException(J9DDRDTFJUtils.java:228)
        at com.ibm.j9ddr.vm29.view.dtfj.java.DTFJJavaObject.fetchDeferredData(DTFJJavaObject.java:610)
        at com.ibm.j9ddr.vm29.view.dtfj.java.DTFJJavaObject.getSize(DTFJJavaObject.java:577)
        at org.eclipse.mat.dtfj.DTFJIndexBuilder.getObjectSize(DTFJIndexBuilder.java:4775)
        at org.eclipse.mat.dtfj.DTFJIndexBuilder.processHeapObject(DTFJIndexBuilder.java:3627)
        at org.eclipse.mat.dtfj.DTFJIndexBuilder.fill(DTFJIndexBuilder.java:2143)
        at org.eclipse.mat.parser.internal.SnapshotFactoryImpl.parse(SnapshotFactoryImpl.java:331)
        at org.eclipse.mat.parser.internal.SnapshotFactoryImpl.openSnapshot(SnapshotFactoryImpl.java:202)
        at org.eclipse.mat.snapshot.SnapshotFactory.openSnapshot(SnapshotFactory.java:149)
        at org.eclipse.mat.ui.snapshot.ParseHeapDumpJob.run(ParseHeapDumpJob.java:97)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: com.ibm.j9ddr.InvalidDataTypeException: U64 contains value larger than Long.MAX_VALUE
        at com.ibm.j9ddr.vm29.types.U64.longValue(U64.java:115)
        at com.ibm.j9ddr.vm29.types.UDATA.mult(UDATA.java:233)
        at com.ibm.j9ddr.vm29.j9.gc.GCArrayletObjectModelBase.externalArrayletsSize(GCArrayletObjectModelBase.java:439)
        at com.ibm.j9ddr.vm29.j9.gc.GCArrayletObjectModel_V2.getTotalFootprintInBytesWithHeader(GCArrayletObjectModel_V2.java:66)
        at com.ibm.j9ddr.vm29.j9.gc.GCObjectModel_V1.getTotalFootprintInBytesWithHeader(GCObjectModel_V1.java:176)
        at com.ibm.j9ddr.vm29.j9.ObjectModel.getTotalFootprintInBytesWithHeader(ObjectModel.java:205)
        at com.ibm.j9ddr.vm29.view.dtfj.java.DTFJJavaObject.fetchDeferredData(DTFJJavaObject.java:608)
        ... 9 more
```

--- a/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModelBase.java
+++ b/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModelBase.java
@@ -438,7 +438,11 @@ public abstract class GCArrayletObjectModelBase extends GCArrayObjectModel
 	{
 		UDATA numberArraylets = numExternalArraylets(array);
 
-		return numberArraylets.mult(arrayletLeafSize);
+		if (numberArraylets.isZero()) {
+			return numberArraylets;
+		} else {
+			return numberArraylets.mult(arrayletLeafSize);
+		}
 	}
 
 }

